ui/src/app/certificate_requests/table.tsx

```diff
@@ -1,99 +1,305 @@
-                            </tr>
-                        </thead>
-                        <tbody>
-                            {
-                                sortedRows().map((row) => (
-                                    <Row key={row.id} id={row.id} csr={row.csr} certificate={row.certificate} ActionMenuExpanded={actionsMenuExpanded} setActionMenuExpanded={setActionsMenuExpanded} />
-                                )
-                                )}
-                        </tbody>
-                        {rows.length == 0 && <EmptyState asideOpen={isAsideOpen} setAsideOpen={setAsideIsOpen} />}
-                    </table>
+    };
+
+    const getExpiryColor = (notAfter?: string): string => {
+        if (!notAfter) return 'inherit';
+        const expiryDate = new Date(notAfter);
+        const now = new Date();
+        const timeDifference = expiryDate.getTime() - now.getTime();
+        if (timeDifference < 0) return "rgba(199, 22, 43, 1)";
+        if (timeDifference < 24 * 60 * 60 * 1000) return "rgba(249, 155, 17, 0.45)";
+        return "rgba(14, 132, 32, 0.35)";
+    };
+
+    const getFieldDisplay = (label: string, field: string | undefined, compareField?: string | undefined) => {
+        const isMismatched = compareField !== undefined && compareField !== field;
+        return field ? (
+            <p style={{ marginBottom: "4px" }}>
+                <b>{label}:</b>{" "}
+                <span style={{ color: isMismatched ? "rgba(199, 22, 43, 1)" : "inherit" }}>
+                    {field}
+                </span>
+            </p>
+        ) : null;
+    };
+
+    const rows = csrs.map((csrEntry) => {
+        const { id, csr, certificate } = csrEntry;
+        const csrObj = extractCSR(csr);
+        const certs = splitBundle(certificate);
+        const clientCertificate = certs?.at(0);
+        const certObj = clientCertificate ? extractCert(clientCertificate) : null;
+
+        const isCSRContentVisible = showCSRContent === id;
+        const isCertContentVisible = showCertContent === id;
+
+        return {
+            sortData: {
+                csr_status: certificate === "" ? "outstanding" : (certificate === "rejected" ? "rejected" : "fulfilled"),
+                cert_expiry_date: certObj?.notAfter || "",
+            },
+            columns: [
+                { content: id.toString() },
+                { content: csrObj.commonName || "N/A" },
+                { content: certificate === "" ? "outstanding" : (certificate === "rejected" ? "rejected" : "fulfilled") },
+                {
+                    content: certObj?.notAfter || "",
+                    style: { backgroundColor: getExpiryColor(certObj?.notAfter) },
+                },
+                {
+                    content: (
+                        <>
+                            {successNotificationId === id && <SuccessNotification successMessage="CSR copied to clipboard" />}
+                            <ContextualMenu
```

Comment:
I hope there's a way to bring the groups back, because without them it's difficult to parse the list of actions. Maybe there are more creative solutions to this.

Reply:
I'm not sure honestly, I haven't seen an easy way to do this

Reply:
I'm ok with doing this in a separate PR

Reply:
Done